Task 16

RuslanGabd opened the pull request.

RuslanGabd committed:
init commit

RuslanGabd committed:
init

RuslanGabd committed:
spring mvc

RuslanGabd committed:
Merge remote-tracking branch 'origin/TASK_16' into TASK_16

# Conflicts:
#	core/pom.xml
#	core/src/main/java/com/ruslan/services/BookService.java
#	core/src/main/java/com/ruslan/services/OrderService.java
#	core/src/main/java/com/ruslan/servlet/BookServlet.java
#	pom.xml
#	ui/pom.xml
#	ui/src/main/java/com/ruslan/console/Builder.java
#	ui/src/main/java/com/ruslan/console/IAction.java
#	ui/src/main/java/com/ruslan/console/Menu.java
#	ui/src/main/java/com/ruslan/console/MenuController.java
#	ui/src/main/java/com/ruslan/console/MenuItem.java
#	ui/src/main/java/com/ruslan/console/Navigator.java
#	ui/src/main/java/com/ruslan/controller/BookController.java
#	ui/src/main/java/com/ruslan/controller/StartController.java

RuslanGabd committed:
spring mvc

RuslanGabd committed:
spring mvc

RuslanGabd committed:
spring mvc

RuslanGabd committed:
spring mvc

RuslanGabd committed:
spring mvc

RuslanGabd committed:
spring mvc

RuslanGabd committed:
spring mvc

RuslanGabd committed:
spring mvc

RuslanGabd committed:
spring mvc

RuslanGabd committed:
spring mvc

RuslanGabd committed:
spring mvc

RuslanGabd committed:
spring mvc

RuslanGabd committed:
spring mvc

RuslanGabd committed:
spring mvc

mrseneca commented on core/src/main/java/com/ruslan/utils/MappingBookToDto.java:
```diff
@@ -0,0 +1,28 @@
+import org.springframework.stereotype.Service;
+
+@Service
+public class MappingBookToDto {
```
Usually this type of classes called mappers, right now it looks like method name :)
```suggestion
public class BookMapper {
```

mrseneca replied:
Also they can have static methods for this just to not add them to context, but it's unnecessary

mrseneca commented on ui/src/main/java/com/ruslan/controller/OrderController.java:
```diff
@@ -0,0 +1,97 @@
+    }
+
+    @DeleteMapping("/{id}")
+    public String deleteOrder(@PathVariable int id) {
```
Better have generic response dto with string value inside it (or Map atleast) than just string, in this way you always will have json as response

mrseneca reviewed (changes requested):
Good job, Few fixes and it will be ready. Also I can't understand what you did with DI and UI modules, because GIT telling you just moved it to different package, but if you have Spring and Rest representation you can get rid of it at all. Please, if i wrong let me know

RuslanGabd committed:
deleted module DI and unused classes.

RuslanGabd commented:
> Good job, Few fixes and it will be ready. Also I can't understand what you did with DI and UI modules, because GIT telling you just moved it to different package, but if you have Spring and Rest representation you can get rid of it at all. Please, if i wrong let me know

Thank you for review! Yes I didn't deleted modules  DI and UI. For now moment I deleted DI and unused classes in Core. I moved components spring-web from UI to new module "springweb".  I would like to leave module UI, if it is possible. Other comments was fixed.

mrseneca commented on springweb/src/main/java/com/ruslan/controller/BookController.java:
```diff
@@ -0,0 +1,86 @@
+    }
+
+    @DeleteMapping("/{id}")
+    public ResponseEntity <String> deleteBook(@PathVariable Integer id) {
```
Under generic response dto I meant dto class with string field inside, not` ResponseEntity<String>`. This will return same string response, but you will be able configure it. I meant somnethink like this:

```
public class GenericResponseDto {
    private String message;

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }
}
```

and then `ResponseEntity<GenericResponseDto>`

RuslanGabd replied:
thanks, I fixed

mrseneca commented on springweb/src/main/java/com/ruslan/controller/BookController.java:
```diff
@@ -0,0 +1,86 @@
+
+
+    @GetMapping("/poor-purchased")
+    public ResponseEntity<List<Book>> getStaleBooks() {
```
Entities must not be returned, but DTOs

mrseneca reviewed (changes requested):
Good. Last things. 
Also why you want UI still be here?

RuslanGabd commented:
> Good. Last things. Also why you want UI still be here?

The UI could help me test  app if I add new features to Core, but it is not important, the UI has been removed. Other comments was fixed

RuslanGabd committed:
deleted module UI and added GenericResponseDto

mrseneca reviewed (approved):
Well done
